### DISCUSSION TITLE
IObjectTypeDescriptor ResolverWith Obtain the Fields Value Only

### DISCUSSION THREAD

**Discussion by OneLogicalMyth** [ChilliCream/hotchocolate#3121] [Q&A] [answered]
Hi there,

Firstly thanks for all the hard work on hot chocolate, it has been a great aid for my project I am currently working on.

I am not a developer but do have experience working with c# and have managed to follow most of the documentation perfectly fine. One thing that has eluded me is how the resolvers work for the descriptor.

Lets say I have this object:

~~~c#
public class MyObject
{
    public int Id { get; set; }
    public string Name { get; set; }

    [TimeStamp]
    public byte[] RowVersion { get; set; }
}
~~~

I then define the object type using code-first as:

~~~c#
public class MyObjectType : ObjectType<MyObject>
{

    protected override void Configure(IObjectTypeDescriptor<MyObject> descriptor)
    {
            descriptor
                .ResolveWith<SomeResolver>(o => o.GetSomething(default));

            // how do I make this work?
            descriptor
                .ResolveWith<SomeResolver>(o => o.GetSomethingFieldValue(default));
    }

   private class SomeResolver
   {
       public string GetSomething(MyObject myObject)
       {
            return myObject.Name;
       }

       // how can I just pass in the value of the field??!
       public string GetSomethingFieldValue(string name)
       {
            return name;
       }
   }


}
~~~

I would love to just pass in the value of the field, then I can make my resolver class more generic. Then I can just have a global resolver that takes an input and gives an output etc. if that makes sense.

**Comment by michaelstaib** [comment, marked as answer]
This is something we are discussing but have not found a nice enough way so far. Sometime you do want to have two fields and sometimes a field completely unrelated to the one you are resolving. We will discuss changes to this for the May release.

**Reply by OneLogicalMyth** [reply]
Thanks for the quick reply, I thought I was miss understanding. So sounds like its something for a future release, I will proceed with a method per class and property for now then as a workaround.
